File: rental-app-flutter/lib/features/auth/models/user_dto.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'user_dto.freezed.dart';
part 'user_dto.g.dart';

@freezed
abstract class UserDto with _$UserDto {
  const factory UserDto({
    required String id,
    required String email,
    required String firstName,
    required String lastName,
    String? phoneNumber,
    String? profilePictureUrl,
    required String role,
    required bool isActive,
    required bool isVerified,
    String? languagePreference,
    String? agencyId,
    String? agencyRole,
    @Default(5.0) double trustScore,
    required String createdAt,
  }) = _UserDto;

  factory UserDto.fromJson(Map<String, dynamic> json) => _$UserDtoFromJson(json);
}

File: rental-backend/src/main/java/com/homeflex/core/dto/response/UserDto.java
package com.homeflex.core.dto.response;

import java.time.LocalDateTime;
import java.util.List;
import java.util.UUID;

public record UserDto(
        UUID id,
        String email,
        String firstName,
        String lastName,
        String phoneNumber,
        String profilePictureUrl,
        String role,           // legacy — first matched enum value; kept for frontend compat
        List<String> roles,    // RBAC role names, e.g. ["ROLE_TENANT"]
        List<String> permissions, // flattened permission names, e.g. ["BOOKING_CREATE", ...]
        Boolean isActive,
        Boolean isVerified,
        String languagePreference,
        UUID agencyId,
        String agencyRole,
        Double trustScore,
        Boolean emailNotificationsEnabled,
        Boolean pushNotificationsEnabled,
        Boolean smsNotificationsEnabled,
        Integer profileCompleteness,
        LocalDateTime createdAt,
        Boolean stripeConnected,
        String stripeAccountId
) {}
